### PR TITLE
Add Azurite Node devcontainer

### PR DESCRIPTION
**Pull request by AlexanderYW**
This is a new devcontainer that setup recommended extensions and ports for running Azurite locally in a devcontainer

It also contains two tests for `blob` and `queue`

### CODE REVIEW

File: containers/javascript-node-azurite/.devcontainer/docker-compose.yml
Comment (ItalyPaleAle) [inline containers/javascript-node-azurite/.devcontainer/docker-compose.yml:30]: Would it make sense to add `container_name: storage` (or `azurite`) so the other services can connect to Azurite via `http://storage:10000` directly?

File: containers/javascript-node-azurite/.devcontainer/docker-compose.yml
Comment (ItalyPaleAle) [inline containers/javascript-node-azurite/.devcontainer/docker-compose.yml:34]: Should the entire `/data` folder be mounted in the container?  As per docs: https://github.com/Azure/Azurite#run-azurite-v3-docker-image-with-customized-persisted-data-location

**Review by ItalyPaleAle — COMMENTED** [review]
I've used Azurite in a dev container before and I think this is a good example, but I've proposed a few tweaks (I like how this should work through Codespaces too!)

File: containers/javascript-node-azurite/.devcontainer/docker-compose.yml
Comment (AlexanderYW) [reply containers/javascript-node-azurite/.devcontainer/docker-compose.yml:34]: Thanks for pointing that out, I will correct that soon :)

File: containers/javascript-node-azurite/.devcontainer/docker-compose.yml
Comment (AlexanderYW) [reply containers/javascript-node-azurite/.devcontainer/docker-compose.yml:30]: Sure, I can add that :)

**Comment by AlexanderYW** [comment]
> I've used Azurite in a dev container before and I think this is a good example, but I've proposed a few tweaks (I like how this should work through Codespaces too!)

To be honest with you, I tried reaching out to a few friends if they had gotten early access to codespaces, so they could test it for me since I haven't, but no one had, if you have, I would love if you could test it out, but based on my assumptions it should 🙃

File: containers/javascript-node-azurite/README.md
Comment (2percentsilk) [inline containers/javascript-node-azurite/README.md:1]:
```suggestion
# Node.js & Azurite (Community)
```

File: containers/javascript-node-azurite/README.md
Comment (2percentsilk) [inline containers/javascript-node-azurite/README.md:10]: We've added a new piece of metadata: Categories! 

```suggestion
| *Categories* | Community, Azure
| *Definition type* | Docker Compose |
```

**Review by 2percentsilk — CHANGES_REQUESTED** [review]
Apologies for the delay here. This addition looks great. I suggested two small changes since we've updated how we title and tag the definitions so you'll need to please accept those.

As the final piece, I want to confirm that you're willing to be the maintainer for this definition - generally this means we'll loop you in for comment/approval for any other PRs that the community may make against this definition. Once you confirm this, I'll be happy to merge!

**Comment by AlexanderYW** [comment]
Sure, I can confirm that I'm willing to maintain this 🙂

**Pull request merged**
